### PR TITLE
Fix test results

### PR DESCRIPTION
**Pull request by vilmakoo**
Handle unsupported exercise types.
Fix error when retrying posting model after template results are received (and exercise.status has changed).

 :metal:

### CODE REVIEW

File: lib/sandbox_results_handler.rb [hunk 1]
```diff
@@ -24,17 +24,20 @@ def generate_data_for_frontend(test_output, package_type)
...
+    end
     error = { header: header, messages: messages }
     @exercise.error_messages.push error
   end
 
   def compile_errors(test_output, package_type)
     return unless test_output['status'] == 'COMPILE_FAILED'
     header = package_type == 'TEMPLATE' ? 'Tehtäväpohja ei kääntynyt: ' : 'Malliratkaisu ei kääntynyt: '
-    messages = error_message_lines(test_output).join('\n')
+    messages = error_message_lines(test_output).join('<rivinvaihto>')
```
Comment (pirtnea) [inline lib/sandbox_results_handler.rb]: :finland: ?

File: lib/test_templates.rb [hunk 2]
```diff
@@ -46,7 +46,7 @@ def string_stdin_string_stdout_test_code
 
       String out = io.getSysOut();
 
-      assertTrue("Kun syöte oli '" + input.replaceAll("\\n", "\\\\n") + "' tulostus oli: '" + out.replaceAll("\\n", "\\\\n") + "', mutta se ei sisältänyt: '" + output.replaceAll("\\n", "\\\\n") + "'.", out.contains(output));
+      assertTrue("Kun syöte oli '" + input.replaceAll("\\n", "\\\\\\n") + "' tulostus oli: '" + out.replaceAll("\\n", "\\\\\\n") + "', mutta se ei sisältänyt: '" + output.replaceAll("\\n", "\\\\\\n") + "'.", out.contains(output));
```
Comment (pirtnea) [inline lib/test_templates.rb]: Excellent backslashing :star2:

File: lib/sandbox_results_handler.rb [hunk 3]
```diff
@@ -24,17 +24,20 @@ def generate_data_for_frontend(test_output, package_type)
 
   def test_results(test_output)
     # Push test results into exercise's error messages
-    return if test_output['testResults'].empty? || test_output['testResults'].first['successful']
+    return if test_output['testResults'].empty? || test_output['status'] == 'PASSED'
     header = 'Virheet testeissä: '
-    messages = test_output['testResults'][0]['message']
+    messages = ''
```
Comment (nygrenh) [inline lib/sandbox_results_handler.rb]:
```ruby
messages = test_output['testResults'].each_with_object('') do |test, string|
  string += messages += "#{test['message']}<rivinvaihto>"
end
```

Comment (pirtnea) [reply lib/sandbox_results_handler.rb]: Good code.

File: lib/sandbox_results_handler.rb (hunk 1 above)
Comment (nygrenh) [reply lib/sandbox_results_handler.rb]: Why add rivinvaihto here?

File: lib/test_templates.rb (hunk 2 above)
Comment (vilmakoo) [reply lib/test_templates.rb]: Thanks :+1:

File: lib/sandbox_results_handler.rb (hunk 1 above)
Comment (vilmakoo) [reply lib/sandbox_results_handler.rb]: 'rivinvaihto' so that crowdsorcerer won't get all confused.

File: lib/sandbox_results_handler.rb (hunk 3 above)
Comment (vilmakoo) [reply lib/sandbox_results_handler.rb]: don't understand the requested change, it doesn't seem to compile.

Comment (nygrenh) [reply lib/sandbox_results_handler.rb]: Should be better now

Comment (vilmakoo) [reply lib/sandbox_results_handler.rb]: can't modify frozen string

Comment (nygrenh) [reply lib/sandbox_results_handler.rb]: Ah, the good ol' magic comment. Maybe try replacing the empty string with `String.new`?

**Pull request merged**